error: ‘isnan’ was not declared in this scope

Hi.

Im getting an error while building the library per instructions.

```
sbr@hiltrud ~/Code/kazmath/build $ make
Scanning dependencies of target kazmath
[  5%] Building C object kazmath/CMakeFiles/kazmath.dir/mat4.c.o
[ 11%] Building C object kazmath/CMakeFiles/kazmath.dir/mat3.c.o
[ 16%] Building C object kazmath/CMakeFiles/kazmath.dir/plane.c.o
[ 22%] Building C object kazmath/CMakeFiles/kazmath.dir/vec4.c.o
[ 27%] Building C object kazmath/CMakeFiles/kazmath.dir/quaternion.c.o
[ 33%] Building C object kazmath/CMakeFiles/kazmath.dir/vec2.c.o
[ 38%] Building C object kazmath/CMakeFiles/kazmath.dir/vec3.c.o
[ 44%] Building C object kazmath/CMakeFiles/kazmath.dir/utility.c.o
[ 50%] Building C object kazmath/CMakeFiles/kazmath.dir/aabb2.c.o
[ 55%] Building C object kazmath/CMakeFiles/kazmath.dir/aabb3.c.o
[ 61%] Building C object kazmath/CMakeFiles/kazmath.dir/ray2.c.o
[ 66%] Building C object kazmath/CMakeFiles/kazmath.dir/ray3.c.o
[ 72%] Building C object kazmath/CMakeFiles/kazmath.dir/GL/mat4stack.c.o
[ 77%] Building C object kazmath/CMakeFiles/kazmath.dir/GL/matrix.c.o
[ 83%] Linking C static library libkazmath.a
[ 83%] Built target kazmath
[ 88%] Generating ../../tests/main.cpp
Scanning dependencies of target kazmath_tests
[ 94%] Building CXX object tests/CMakeFiles/kazmath_tests.dir/main.cpp.o
In file included from /home/sbr/Code/kazmath/tests/main.cpp:5:0:
/home/sbr/Code/kazmath/tests/test_quaternion.h: In member function ‘void TestQuaternion::test_quaternion_slerp_edge_case()’:
/home/sbr/Code/kazmath/tests/test_quaternion.h:33:36: error: ‘isnan’ was not declared in this scope
         assert_true(!isnan(result.x));
                                    ^
/home/sbr/Code/kazmath/tests/kaztest/kaztest.h:15:43: note: in definition of macro ‘assert_true’
 #define assert_true(actual) _assert_true((actual), __FILE__, __LINE__)
                                           ^
/home/sbr/Code/kazmath/tests/test_quaternion.h:33:36: note: suggested alternative:
         assert_true(!isnan(result.x));
                                    ^
/home/sbr/Code/kazmath/tests/kaztest/kaztest.h:15:43: note: in definition of macro ‘assert_true’
 #define assert_true(actual) _assert_true((actual), __FILE__, __LINE__)
                                           ^
In file included from /usr/include/c++/5/random:38:0,
                 from /usr/include/c++/5/bits/stl_algo.h:66,
                 from /usr/include/c++/5/algorithm:62,
                 from /home/sbr/Code/kazmath/tests/kaztest/kaztest.h:9,
                 from /home/sbr/Code/kazmath/tests/main.cpp:5:
/usr/include/c++/5/cmath:641:5: note:   ‘std::isnan’
     isnan(_Tp __x)
     ^
tests/CMakeFiles/kazmath_tests.dir/build.make:76: recipe for target 'tests/CMakeFiles/kazmath_tests.dir/main.cpp.o' failed
make[2]: *** [tests/CMakeFiles/kazmath_tests.dir/main.cpp.o] Error 1
CMakeFiles/Makefile2:140: recipe for target 'tests/CMakeFiles/kazmath_tests.dir/all' failed
make[1]: *** [tests/CMakeFiles/kazmath_tests.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
```

Im using gcc 5.4.0. on Ubuntu.

--- a/tests/test_quaternion.h
+++ b/tests/test_quaternion.h
@@ -1,4 +1,4 @@
-#include <math.h>
+#include <cmath>
 
 #include "kaztest/kaztest.h"
 #include "../kazmath/quaternion.h"
@@ -32,10 +32,10 @@ public:
         kmQuaternion result;
         kmQuaternionSlerp(&result, &initial, &dest, step);
 
-        assert_true(!isnan(result.x));
-        assert_true(!isnan(result.y));
-        assert_true(!isnan(result.z));
-        assert_true(!isnan(result.w));
+        assert_true(!std::isnan(result.x));
+        assert_true(!std::isnan(result.y));
+        assert_true(!std::isnan(result.z));
+        assert_true(!std::isnan(result.w));
     }
 
     void test_quaternion_get_up_vector() {

